solving-coding-challenges/solving-coding-challenges.md

```diff
@@ -0,0 +1,190 @@
+	- Test your solution
+
+### Materials
+- [How to think like a programmer](https://www.freecodecamp.org/news/how-to-think-like-a-programmer-lessons-in-problem-solving-d1d8bf1de7d2/) - High level overview on how to approach coding problems
+- [10 Steps to Solving a Programming Problem](https://codeburst.io/10-steps-to-solving-a-programming-problem-8a32d1e96d74) - Another article with steps on how to solve programming problems, with an example
+
+
+### Lesson
+__Solving the problem__
+
+**1.Understand the prompt**
+
+Take time to read the problem and understand what the goals are. If test cases are given, look at each one and understand them. 
+	
+**2.Think of a high level approach**
+
+Based on your learning style, you might work to understand the problem in different ways. For example, drawing a diagram might help some people. Talking and explaining out loud might help other people. 
+
+ At the end of this step, you should be able to explain your solution (without using code). 
+ 
+**3. Code your solution**
+
+Begin coding! Debugging and console logging along the way can help you catch errors early, and understand how your code is working. Try to break down the problem into sub problems, and tackle those one at a time. Some things to think about are:
+- What format is the input in? What format do I want the output to be?
+- What information do I need to store? What information do I need to calculate?
+
+**4. Test your solution**
+ See [Testing lesson](https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/testing-mindset.md)
+
+---------
+
+__Getting Stuck__
+
+_Debugging_
+
+Testing along the way can make debugging easier. See [Debugging Lesson](https://github.com/Techtonica/curriculum/blob/master/debugging/debugging.md) for more info. 
+
+_Using Google/StackOverflow_
+
+ If you're stuck, Google and StackOverflow have a ton of information, which has pros and cons. For any question or topic, there will usually be a snippet of code or example somewhere.  While it is tempting to copy and paste these into coding challenges, there are disadvantages such as:
+- By copy-pasting, you lose the chance to problem solve yourself
+- Often these code snippets might not be 100% what you're looking for, so they could introduce more bugs
+
+When you do Google, try to have a specific goal in mind. ie, "I want to look up the syntax for arrays", or "I want to see an example of recursion". 
+
+
+### Common Mistakes & Misconceptions
+- **I can't test until I finish**
+
+Testing and running your code along the way is important because it can tell you if errors or bugs pop up along the way. If you wait until finishing it to test, you might have to go back and redo things that you could have caught earlier on.
+
+- **I can't ask for help** 
+
+ Being able to ask for help effectively is an important and crucial skill. In software engineering no one knows anything, so knowing how to problem solve yourself and when and how to ask for help is important. 
```

Comment:
No one knows *everything*, maybe?

Reply:
haha oops yes that's what it was supposed to say 😆